### PR TITLE
common: Fix clang compilation

### PR DESCRIPTION
**Pull request by byo**
Signed-off-by: Bartłomiej Święcki <bartlomiej.swiecki@corp.ovh.com>

### CODE REVIEW

**Comment by Liuchang0812** [comment]
hi, could we merge it now? I need this patch.

File: src/common/perf_histogram.h [hunk 1]
```diff
@@ -76,8 +76,10 @@ class PerfHistogram : public PerfHistogramCommon {
   }
 
   /// Copy from other histogram object
-  PerfHistogram(const PerfHistogram &other)
-      : m_axes_config(other.m_axes_config) {
+  PerfHistogram(const PerfHistogram &other) {
+    for (int i = 0; i < DIM; i++) {
```
Comment (tchaikov) [inline src/common/perf_histogram.h]: could use `memcpy()` for better performance.

**Comment by tchaikov** [comment]
@Liuchang0812 it needs an approval and a qa run.

**Comment by Liuchang0812** [comment]
@tchaikov OK

File: src/common/perf_histogram.h (hunk 1 above)
Comment (tchaikov) [inline src/common/perf_histogram.h]: @byo on a second thought, we can also leverage `std::array` for this
```patch
diff --git a/src/common/perf_histogram.h b/src/common/perf_histogram.h
index 6d315905db..99274d765d 100644
--- a/src/common/perf_histogram.h
+++ b/src/common/perf_histogram.h
@@ -20,6 +20,7 @@
 #include "include/int_types.h"

 #include <memory>
+#include <array>

 class PerfHistogramCommon {
 public:
@@ -133,7 +134,7 @@ protected:
   std::unique_ptr<atomic64_t[]> m_rawData;

   /// Configuration of axes
-  axis_config_d m_axes_config[DIM];
+  std::array<axis_config_d,DIM> m_axes_config;

   /// Dump histogram counters to a formatter
   void dump_formatted_values(ceph::Formatter *f) const {
@@ -145,8 +146,8 @@ protected:
   /// Get number of all histogram counters
   int64_t get_raw_size() {
     int64_t ret = 1;
-    for (int i = 0; i < DIM; ++i) {
-      ret *= m_axes_config[i].m_buckets;
+    for (const auto &ac : m_axes_config) {
+      ret *= ac.m_buckets;
     }
     return ret;
   }
```

Comment (byo) [inline src/common/perf_histogram.h]: @tchaikov Great idea! I guess that's what std::array was created for. And I'm almost sure that copying such array to another one would be automatically converted to memcpy if the element type is trivially copyable.

Will give it a try in a second.

**Comment by byo** [comment]
@tchaikov Changed as you suggested, looks much better now, also added small unit test to ensure the copy works correctly.

**Comment by byo** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-02-17_10:27:51-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2017-02-18_07:46:40-rados-wip-kefu-testing---basic-mira/

**Pull request merged**